app/Services/Subusers/SubuserCreationService.php

```diff
@@ -57,10 +58,12 @@ public function handle(Server $server, string $email, array $permissions): Subus
                 throw new ServerSubuserExistsException(trans('exceptions.subusers.subuser_exists'));
             }
 
+            $cleanedPermissions = array_filter(array_unique($permissions), fn ($permission) => $permission === Permission::ACTION_WEBSOCKET_CONNECT || auth()->user()->can($permission, $server));
+
```

Comment:
```suggestion
            $cleanedPermissions = collect($permissions)
                ->unique()
                ->filter(fn ($permission) => auth()->user()->can($permission, $server))
                ->sort()
                ->values()
                ->all();
```
Otherwise we get `{"4":"user.read","5:"user.create","6:"user.update,"7:"user.delete"` instead of `["user.create","user.delete","user.read","user.update"]`

Also why should a user that can't connect to the WEBSOCKET be able to grant that ?